bot.py

```diff
@@ -151,6 +151,32 @@ def real_decorator(func):
         parameters = ' '.join(message.content.strip().lower().split(' ')[1:])
         await parse_command(command, message, parameters)
 
+@client.event
+async def on_member_remove(member):
+    member_id = member.id
+    if member_id in session[1]:
+        if session[0] and session[1][member_id][0] or not session[0]:
```

Comment:
nit: You can move the `not session[0]` case into an elif block so you don't have to check session[0] again